src/parsers.rs

```diff
@@ -181,31 +181,28 @@ named!(sign <i32>, alt!(
         )
     );
 
-named!(timezone_hour <(i32,i32)>, chain!(
-        s: sign ~
-        h: hour ~
+named!(timezone_hour <(i32,i32)>, do_parse!(
+        s: sign >>
+        h: hour >>
         m: empty_or!(
-            chain!(
-                tag!(":")? ~ m: minute , || { m }
-            ))
-        ,
-        || { (s * (h as i32) , s * (m.unwrap_or(0) as i32)) }
+            do_parse!(opt!(tag!(":")) >> m: minute >> ( m ))
```

Comment:
(then again the `do_parse .. >> ... >>` seems to be easier to read than nested calls to preceded, map, etc.